### ISSUE TITLE
[Bug]: Actively generated `request` is starved when new requests arrive (tensor parallel)

### ISSUE THREAD

**Issue by llmwiz** [vllm-project/vllm#11067]
### Your current environment

<details>
<summary>The output of `python collect_env.py`</summary>

```text
Is CUDA available: True
CUDA runtime version: 12.0.140
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration:
GPU 0: NVIDIA GeForce RTX 4090
GPU 1: NVIDIA GeForce RTX 4090
GPU 2: NVIDIA GeForce RTX 4090
GPU 3: NVIDIA GeForce RTX 4090

Nvidia driver version: 550.127.05
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                         x86_64
Model name:                           AMD EPYC

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] nvidia-cublas-cu12==12.4.5.8
[pip3] nvidia-cuda-cupti-cu12==12.4.127
[pip3] nvidia-cuda-nvrtc-cu12==12.4.127
[pip3] nvidia-cuda-runtime-cu12==12.4.127
[pip3] nvidia-cudnn-cu12==9.1.0.70
[pip3] nvidia-cufft-cu12==11.2.1.3
[pip3] nvidia-curand-cu12==10.3.5.147
[pip3] nvidia-cusolver-cu12==11.6.1.9
[pip3] nvidia-cusparse-cu12==12.3.1.170
[pip3] nvidia-ml-py==12.560.30
[pip3] nvidia-nccl-cu12==2.21.5
[pip3] nvidia-nvjitlink-cu12==12.4.127
[pip3] nvidia-nvtx-cu12==12.4.127
[pip3] pyzmq==26.2.0
[pip3] torch==2.5.1
[pip3] torchvision==0.20.1
[pip3] transformers==4.47.0
[pip3] triton==3.1.0
[conda] Could not collect (using venv instead)
ROCM Version: Could not collect
Neuron SDK Version: N/A
vLLM Version: 0.6.4.post1
vLLM Build Flags:
CUDA Archs: Not Set; ROCm: Disabled; Neuron: Disabled
GPU Topology: (PCI-e - all GPUs on the same mobo)
```

</details>

### 🐛 Describe the bug

Please help me understand problematic scheduler behavior for vLLM openAI API server:
`vllm serve Qwen/Qwen2.5-72B-Instruct-GPTQ-Int4 --max-seq_len-to-capture 32768 --max-model-len 32768 -tp 4`

1. Request `A` comes in (prompt: "tell me super long story") and starts being generated immediately.
2. Request `B` arrives (30k tokens prompt), immediately causing Request `A` to pause generation.
3. Once request `B` is fully prompt-processed, token generation resumes for request `A` and begins for request `B`.

In other words, prompt-processing always takes precedence over token-generation within `--max-num-seqs` pool. This way it's easy to construct a scenario where request `A` is being starved for quite long. If I understand, the rationale behind this behavior is that prompt-processing fully occupies all `-tp N` GPUs for only one request, while token-generation (sequential in nature) can perform N requests well in parallel. So vLLM tries to feed GPUs with N requests at the time. While this works efficiently for offline generation, it may be problematic for online use case where response generation for one user is paused several times before completion.

Is my understanding correct that vLLM in tensor-parallel mode is incapable of "token generation" and "prompt processing" for two requests at the same time?

(Also, I can workaround this by setting `--max-num-seqs` to `1`, but that's far from optimal.)

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.
